src/Value/Color.php

```diff
@@ -256,4 +250,23 @@ private function allComponentsAreNumbers(): bool
 
         return true;
     }
+
+    /**
+     * Note that this method assumes the following:
+     * - The {@see aComponents} array has keys for `r`, `g` and `b`;
+     * - The values in the array are all instances of {@see Size}.
+     *
+     * Errors will be triggered or thrown if this is not the case.
+     */
+    private function renderAsHex(): string
+    {
+        $result = \sprintf(
+            '%02x%02x%02x',
+            $this->aComponents['r']->getSize(),
+            $this->aComponents['g']->getSize(),
+            $this->aComponents['b']->getSize()
+        );
+        return '#' . (($result[0] == $result[1]) && ($result[2] == $result[3]) && ($result[4] == $result[5])
```

Comment:
While we're at it, I propose adding a local variable `$canUseShorthand`.

Reply:
Done.  I also changed the expression to avoid variable substitution within a string.